classes/opponentPhaseRolling.lua

```diff
@@ -0,0 +1,74 @@
+-- The opponent phase with normal gravity
+
+local opponentPhaseRolling = {}
+opponentPhaseRolling.__index = opponentPhaseRolling
+
+
+--------------------
+-- MAIN CALLBACKS --
+--------------------
+
+function opponentPhaseRolling.new()
+	local self = classes.opponentBase.new()
+	setmetatable(self, opponentPhaseRolling)
+
+	-- Constants
+	self.MOVE_SPEED = 200
+	self.BULLET_COOLDOWN_LENGTH = 1
+	GRAVITY = 250
+
+	-- Movement variables
+	self.direction = false
```

Comment:
Rather than a `self.direction` boolean, make a `self.xspeed` variable and set it to the move speed. Then, negate this variable when it hits the wall (or do `math.abs` shenanigans like I did)